File: 424-longest-repeating-character-replacement/424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int n = s.length();
        int count[] = new int[26];
        int i=0;
        int maxCount = 0;
        int res = 0;
        for(int j=0;j<n;j++){
            maxCount = Math.max(maxCount, ++count[s.charAt(j)-'A']);
            while(i<= j && j-i+1 - maxCount > k){
                count[s.charAt(i)-'A']--;
                i++;
            }
            res = Math.max(res, j-i+1);
        }
        return res;
    }
}

File: 424-longest-repeating-character-replacement/424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int count[26]={0};
        int n = s.length();
        int max_count = 0;
        int res = 0;
        int i = 0;
        for(int j=0;j<n;j++){
            max_count = max(max_count, ++count[s[j]-'A']);
            while(i < j && (max_count+k) < j-i+1){
                --count[s[i]-'A'];
                i++;
            }
            res = max(res, j-i+1);
        }
        return res;
    }
};